CONTRIBUTING.md

```diff
@@ -1,225 +1,245 @@
 
-- If you are running `make run`, change the `gate_client` section in `backend/cmd/server/repository/conf/deployment.yaml`:
-- If you are running the backend server manually, change the `gate_client` section in `<THUNDER_HOME>/repository/conf/deployment.yaml`:
+---
 
-```yaml
-gate_client:
-  port: 5190
-  scheme: "https"
-  login_path: "/signin"
-```
+## **✨ Making Improvements**
 
-2. Add the local development origin of the Thunder Gate application (https://localhost:5190) to the CORS allowed origins in `<THUNDER_HOME>/repository/conf/deployment.yaml`.
+Have an idea for a small improvement? Use the [Fast Track process](#-fast-track-contributions).
 
-```yaml
-cors:
-  allowed_origins:
-    - "https://localhost:5190"
-```
+**Good improvements include**:
 
-3. Run the Thunder Gate application.
+* Better error messages
+* Input validation enhancements
+* Logging improvements
+* Code readability refactoring
+* Performance optimizations
 
-```bash
-cd frontend
-pnpm --filter @thunder/gate dev
-```
+---
 
-### Setting up the Thunder Develop Application
+## **💡 Sharing Ideas & Defining Problems**
 
-💡 **IMPORTANT**: This section assumes that you have already created the `Develop` application using the initial data setup script. If not, please refer to the [Seed Data](#seed-data-optional) section above.
+Have an idea or problem to share? Want to propose a new feature?
 
-1. First, retrieve the application ID of the **Develop** application from the Thunder server. This will be the application with the `client_id` **DEVELOP**.
+**This is where all features start!**
 
-```bash
-curl -k -X GET "https://localhost:8090/applications"
-```
+You can either:
+* **Start a discussion** if your idea needs exploration and community input
+* **Create an issue directly** if you have a clear, well-defined feature request
 
-2. Then, get the current **Develop** application configuration:
+**Choose your path**:
 
-```bash
-curl -k -X GET "https://localhost:8090/applications/<develop-application-id>"
-```
+| Your Situation | Action |
+|---------------|--------|
+| Idea needs refinement | Create a [Discussion](https://github.com/asgardeo/thunder/discussions) |
+| Problem is clear | Create a [Feature Issue](https://github.com/asgardeo/thunder/issues/new/choose) |
+| Not sure? | Start with discussion |
 
-💡 **NOTE**: Replace `<develop-application-id>` with the actual application ID (e.g., `6100bc91-ba99-4ce9-87dd-6d4d80178c38`) obtained from the previous step. The `-k` flag allows curl to work with self-signed SSL certificates in development.
+**→ Learn more**: [Problem Definition Guide](docs/contributing/problem-definition.md)
 
-2. Copy the response from `step 2` and update the `redirect_uris` in the JSON object to include the local development URL (ex: https://localhost:5191/develop). Locate the `inbound_auth_config > config` section and modify the `redirect_uris` array:
+**You can stop here and let someone else design/implement the solution!**
```

Comment:
Shall we remove this?

To me, this sounds bit discouraging to proceed with next steps. Thought about rewording this, but since we mention you can only stop here, in different places, we can remove this part altogether.